cli/cmd/deploy/deploy.go

```diff
@@ -358,6 +388,169 @@ func loginWithTelemetry(ctx context.Context, ch *cmdutil.Helper, redirectURL str
+		if err != nil {
+			return fmt.Errorf("failed to get worktree: %w", err)
+		}
+
+		// git add .
+		if err := wt.AddWithOptions(&git.AddOptions{All: true}); err != nil {
+			return fmt.Errorf("failed to add files to git: %w", err)
+		}
+
+		// git commit -m
+		_, err = wt.Commit("Auto committed by Rill", &git.CommitOptions{All: true})
+		if err != nil {
+			if !errors.Is(err, git.ErrEmptyCommit) {
+				return fmt.Errorf("failed to commit files to git: %w", err)
+			}
+		}
+
+		// Create the remote
+		_, err = repo.CreateRemote(&config.RemoteConfig{Name: "origin", URLs: []string{*githubRepository.HTMLURL}})
+		if err != nil {
+			return fmt.Errorf("failed to create remote: %w", err)
+		}
+
+		// push the changes
+		if err := repo.PushContext(ctx, &git.PushOptions{Auth: &http.BasicAuth{Username: "x-access-token", Password: pollRes.AccessToken}}); err != nil {
+			return fmt.Errorf("failed to push to remote %q : %w", *githubRepository.HTMLURL, err)
+		}
+
+		printer.ColorGreenBold.Printf("\nRepository %q created successfully. Local changes pushed to remote.\n\n", *githubRepository.Name)
+		return nil
+	}
+}
+
+func createGithubRepository(ctx context.Context, ch *cmdutil.Helper, pollRes *adminv1.GetGithubUserStatusResponse, localGitPath, repoOwner string) (*github.Repository, error) {
+	githubClient := github.NewTokenClient(ctx, pollRes.AccessToken)
+
+	defaultBranch := "main"
+	if repoOwner == pollRes.Account {
+		repoOwner = ""
+	}
+	repoName := filepath.Base(localGitPath)
+
+	var githubRepo *github.Repository
+	var err error
+	for i, tempRepoName := 1, repoName; i <= 10; i++ {
+		githubRepo, _, err = githubClient.Repositories.Create(ctx, repoOwner, &github.Repository{Name: &tempRepoName, DefaultBranch: &defaultBranch})
+		if err == nil {
+			break
+		}
+
+		if strings.Contains(err.Error(), "authentication") || strings.Contains(err.Error(), "credentials") {
+			// The users who installed app before we started including repo:write permissions need to accept permissions
+			// and then only we can create repositories.
+			return nil, fmt.Errorf("rill app does not have permissions to create github repository. Visit `https://github.com/settings/installations` to accept new permissions or reinstall app and try again")
+		}
+
+		if !strings.Contains(err.Error(), "name already exists") {
+			return nil, fmt.Errorf("failed to create repository: %w", err)
+		}
+		// there is a name conflict
+		tempRepoName = repoName + fmt.Sprintf("_v%v", i)
+	}
+
+	// the create repo API does not wait for repo creation to be fully processed on server. Need to verify by making a get call in a loop
+	if repoOwner == "" {
+		repoOwner = pollRes.Account
+	}
+
+	ch.Print("\nRequest submitted for creating repository. Checking completion status\n")
+	pollCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+	for {
+		select {
+		case <-pollCtx.Done():
+			return nil, pollCtx.Err()
+		case <-time.After(5 * time.Second):
+			// Ready to check again.
```

Comment:
Can we check more often than every 5 seconds? Maybe every 2 seconds. To make it feel snappier.